opcua/server/address_space.py

```diff
@@ -180,34 +180,46 @@ def add_nodes(self, addnodeitems, user=User.Admin):
     def _add_node(self, item, user):
         result = ua.AddNodesResult()
 
-        if item.RequestedNewNodeId in self._aspace:
+        # Check if the null NodeId is given
+        if item.RequestedNewNodeId.is_null():
+            self.logger.debug("RequestedNewNodeId is null")
+            if item.BrowseName.NamespaceIndex:
```

Comment:
You are right!!!!!  We need a way to specify namespace.... Must be a
solution.... But using namespace of browse name is wrong... Any suggestion?
Going skiing now :-)

On Wed, Mar 23, 2016, 10:05 Denis Štogl <notifications@github.com> wrote:

> In opcua/server/address_space.py
> <https://github.com/FreeOpcUa/python-opcua/pull/149#discussion_r57126962>:
>
> > @@ -180,34 +180,46 @@ def add_nodes(self, addnodeitems, user=User.Admin):
> >      def _add_node(self, item, user):
> >          result = ua.AddNodesResult()
> >
> > -        if item.RequestedNewNodeId in self._aspace:
> > +        # Check if the null NodeId is given
> > +        if item.RequestedNewNodeId.is_null():
> > +            self.logger.debug("RequestedNewNodeId is null")
> > +            if item.BrowseName.NamespaceIndex:
>
> OK, but I would like just to remark something...
>
> This will imply bigger changes since in the test it is written that you
> create NodeId with the NamespaceIndex 4, but since the parent is Objects
> folder you will add it into NamespaceIndex 0 (which is by the way forbidden
> by the standard). So if we always use the NamespaceIndex of the parent
> there is no way to have the Nodes in other NamespaceIndex than 0, and you
> can not specify other NamespaceIndex for the child node (which should be
> possible)....
>
> Maybe I understood you wrong, please correct me...
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/FreeOpcUa/python-opcua/pull/149/files/a6be8b95b418d4abec719247991a80265bbcec86#r57126962>
>